plugins/transformer/basic/client/recordupdater/record_updater.go

```diff
@@ -102,25 +131,43 @@ func (r *RecordUpdater) AddTimestampColumn(columnName string, position int) (arr
 }
 
 func (r *RecordUpdater) ObfuscateColumns(columnNames []string) (arrow.Record, error) {
-	colIndex, err := r.colIndicesByNames(columnNames)
+	plainCols, jsonCols := r.splitJSONColumns(columnNames)
+
+	plainColIndex, err := r.colIndicesByNames(plainCols)
 	if err != nil {
 		return nil, err
 	}
+	jsonColIndex := r.jsonColIndicesByNames(jsonCols)
 
 	oldRecord := r.record.Columns()
 	newColumns := make([]arrow.Array, 0, len(oldRecord))
 	for i, column := range oldRecord {
-		if _, ok := colIndex[i]; ok {
+		if _, ok := plainColIndex[i]; ok {
 			if column.DataType().ID() != arrow.STRING {
 				return nil, fmt.Errorf("column %v is not a string column", r.record.ColumnName(i))
 			}
 			newColumns = append(newColumns, r.obfuscateColumn(column))
-		} else {
+			continue
+		}
+
+		var found bool
+		for jc := range jsonColIndex {
+			if jc.columnName == r.record.ColumnName(i) {
+				if _, ok := column.DataType().(*types.JSONType); !ok {
+					return nil, fmt.Errorf("column %v is not a JSON column", r.record.ColumnName(i))
+				}
+				newColumns = append(newColumns, r.obfuscateJSONColumn(column, jc))
```

Comment:
Looks like `obfuscateJSONColumn` creates a new Arrow JSON array builder per JSON column path. Instead could we create the Arrow JSON array builder per JSON column instead (in case we need to operate on that column), then iterate over the column paths? That should save some allocations I think

Reply:
OK, we now allocate 1 JSON builder per column.